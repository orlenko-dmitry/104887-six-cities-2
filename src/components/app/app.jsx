import React, {Fragment} from 'react';
import {
  arrayOf,
  shape,
} from 'prop-types';

import MainPage from '../main-page/main-page.jsx';
import MainEmptyPage from '../main-empty-page/main-empty-page.jsx';
import withEmptyPage from '../../hocs/withEmptyPage/withEmptyPage.jsx';
import DetailsPage from '../details-page/details-page.jsx';

const WithEmptyPage = withEmptyPage(MainPage, MainEmptyPage);
const getPageScreen = (offers, nearOffers, reviews) => {
  const city = offers[0].city;

  switch (location.pathname) {
    case `/`:
      return <WithEmptyPage />;
    case `/details`:
      return (
        <DetailsPage
          offer={offers[0]}
          reviews={reviews}
          city={city}
        />
      );
  }
  return null;
};

const App = ({offers, nearOffers, reviews}) => (
  <Fragment>{getPageScreen(offers, nearOffers, reviews)}</Fragment>
);

App.propTypes = {
  offers: arrayOf(shape({})),
  nearOffers: arrayOf(shape({})),
  reviews: arrayOf(shape({})),
};

App.defaultProps = {
  offers: [],
  nearOffers: [],
  reviews: [],
};

export default App;
