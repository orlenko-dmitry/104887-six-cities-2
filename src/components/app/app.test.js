import React from 'react';
import {shallow} from 'enzyme';

import {App} from './app.jsx';
import {ASYNC_STATUSES} from '../../consts/index.js';


it(`App renders correctly`, () => {
  const tree = shallow(
      <App
        fetchStatus={ASYNC_STATUSES.SUCCESS}
        getOffers={() => {}}
      />
  );

  expect(tree).toMatchSnapshot();
});
