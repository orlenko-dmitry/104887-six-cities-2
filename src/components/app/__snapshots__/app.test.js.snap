// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Fragment>
  <Connect(Component) />
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    role="alert"
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
</Fragment>
`;
